### ISSUE TITLE
AddRange method throw exception (casting issue)

### ISSUE THREAD

**Issue by Gounlaf** [stewienj/SwordfishCollections#16]
Hi,

I'm trying to use the `AddRange` method of `ConcurrentObservableDictionary`, but it always fail, due to a casting.

```csharp
Unhandled exception. System.InvalidCastException: Unable to cast object of type 'System.Collections.Generic.Dictionary`2[System.String,System.String]' to type 'System.Collections.IList'.
   at Swordfish.NET.Collections.ConcurrentObservableDictionary`2.<>c__DisplayClass4_0.<AddRange>b__2(Int32 index)
   at Swordfish.NET.Collections.ConcurrentObservableBase`2.BodyReadWriteNotify[TRead](TRead readValue, Func`2 write, Func`2[] changes)
   at Swordfish.NET.Collections.ConcurrentObservableBase`2.DoReadWriteNotify[TRead](Func`1 read, Func`2 write, Func`2[] changes)
   at Swordfish.NET.Collections.ConcurrentObservableDictionary`2.AddRange(IEnumerable`1 pairs)
   at MyMethodCallingAddRange() in /where/my/code/is.cs:line 42
```

I tried with some variant of a `IDictionnary` (normal, default concurrent, concurrent from your library).

Temporary fix: I call `AddRange` with a `IList<KeyValuePair<string, string>>`

**Comment by stewienj** [comment]
Hi @Gounlaf, sorry I missed the notification in my inbox last week. I'll look at it now.

**Comment by stewienj** [comment]
New version has been published on nuget here https://www.nuget.org/packages/Swordfish.NET.CollectionsV3/3.3.2

**Comment by stewienj** [comment]
Just found out I've broken some downstream code with the changes I made to this.

**Comment by stewienj** [comment]
Thanks for reporting this issue, it brought to my attention a large blindspot in my unit tests, and allowed me to discover and fix quite a few bugs related to the CollectionChanged event and the bulk add/remove methods on the collection classes.

The new version has been automatically published on nuget (via github actions) here:
https://www.nuget.org/packages/Swordfish.NET.CollectionsV3/3.3.3

Hopefully this won't cause anybody problems downstream.

**Comment by stewienj** [comment]
Any more problems related to the fix for this issue should go into a new issue, as the original problem raised in this issue has now been fixed with the version 3.3.3 release.
